frame/mosaic-vault/src/lib.rs

```diff
@@ -0,0 +1,904 @@
+
+			Self::only_supported_remote_token(remote_network_id.clone(), asset_id.clone())?;
+
+			ensure!(Self::last_transfer(&sender).checked_add(Self::transfer_lockup_time()).ok_or(Error::<T>::Overflow)? < T::BlockTimestamp::now().as_secs(), Error::<T>::TransferNotPossible);
+           
+			ensure!(transfer_delay >= <MinTransferDelay<T>>::get(), Error::<T>::TransferDelayBelowMinimum);
+		
+			ensure!(transfer_delay <= <MaxTransferDelay<T>>::get(), Error::<T>::TransferDelayAboveMaximum);
+
+			ensure!(amount <= Self::max_asset_transfer_size(asset_id), Error::<T>::AmountAboveMaxAssetTransferSize);
+
+			ensure!(amount >= Self::min_asset_transfer_size(asset_id), Error::<T>::AmountBelowMinAssetTransferSize);
+			// 
+			let pallet_account_id = Self::account_id();            
+            // move funds to pallet amount
+		     T::Currency::burn_from(asset_id, &sender, amount).map_err(|_|Error::<T>::BurnFromFailed)?;
+
+			 Self::increase_total_value_transferred(asset_id, amount)?;
+
+			// update in_transfer_funds
+			let in_transfer_funds = Self::in_transfer_funds(asset_id);
+			let new_in_transfer_funds = in_transfer_funds.checked_add(&amount).ok_or(Error::<T>::Overflow)?;
+			<InTransferFunds<T>>::insert(asset_id, new_in_transfer_funds);
+           
+			<LastTransfer<T>>::insert(&sender, T::BlockTimestamp::now().as_secs());
+
+			let deposit_id = Self::generate_deposit_id(remote_network_id, &destination_address, pallet_account_id);
+            <Deposits<T>>::insert(asset_id, DepositInfo{asset_id, amount});
+
+			Self::deposit_event(Event::DepositCompleted{
+				sender,
+				asset_id,
+				remote_asset_id: Self::remote_asset_id(remote_network_id, asset_id),
+				remote_network_id,
+				destination_address,
+				amount, 
+				deposit_id,
+				transfer_delay
+			});
+
+			Ok(().into())
+		 }
+
+		 #[pallet::weight(10_000)]
+		 pub fn withdraw(
+			origin: OriginFor<T>, 
+			destination_account: T::AccountId,
+			amount: T::Balance,
+			asset_id: T::AssetId, 
+			remote_network_id: T::RemoteNetworkId,
+	        deposit_id: T::DepositId,
+			fee: T::Balance,
+		 ) -> DispatchResultWithPostInfo {
+
+			 let sender = ensure_signed(origin.clone())?;
+
+			 T::RelayerOrigin::ensure_origin(origin)?;
+         
+			 ensure!(Self::pause_status() == false, Error::<T>::ContractPaused);
+
+			 Self::only_supported_remote_token(remote_network_id.clone(), asset_id.clone())?;
+             
+			ensure!(Self::has_been_withdrawn(deposit_id) == false, Error::<T>::AlreadyWithdrawn);
+
+			ensure!(Self::get_current_token_liquidity(asset_id)? >= amount, Error::<T>::InsufficientAssetBalance); 
+
+			  <HasBeenWithdrawn<T>>::insert(deposit_id, true);
+
+			  <LastWithdrawID<T>>::put(deposit_id);
+
+			  let pallet_account_id = Self::account_id(); 
+	
+			  let withdraw_amount = amount.saturating_sub(fee);
+
+			 T::Currency::mint_into(asset_id, &pallet_account_id, amount).map_err(|_|Error::<T>::MintToFailed)?;
```

Comment:
mint fee only

Reply:
not sure what you mean here, can you explain